lib/board.rb

```diff
@@ -1,11 +1,106 @@
 class Board
 
+  attr_reader :size, :grid
+
+  FIRST_ELEMENT = 0
+  NUMBER_ONE = 1
+  NUMBER_TWO = 2
+
   def initialize(size)
     @size = size
     @grid = create_grid
   end
 
+  public
+
   def create_grid
-    (0..@size - 1).to_a
+    (NUMBER_ONE..@size).to_a
+  end
+
+  def mark_board(number, mark)
+    @grid[number - NUMBER_ONE] = mark
+  end
+
+  def possible_rows
+    rows = []
+    position = FIRST_ELEMENT
+    until rows.count == dimension
+      rows << @grid.slice(position, dimension)
+      position += dimension
+    end
+    rows
+  end
+
+  def possible_columns
+    columns = []
+    position = FIRST_ELEMENT
+    until columns.count == dimension
+      columns << add_column(position)
+      position += NUMBER_ONE
+    end
+    columns
+  end
+
+  def possible_diagonals
+    diagonals = []
+    position = FIRST_ELEMENT
+    diagonals << add_diagonal(position, :+)
+    diagonals << add_diagonal(position + dimension - NUMBER_ONE, :-)
+    diagonals
+  end
+
+  def dimension
+    Math.sqrt(size)
+  end
+
+  def win?
+    position = FIRST_ELEMENT
+    array_of_results = []
+    until position == number_of_possible_combinations || array_of_results.include?(true)
```

Comment:
until the position is equal to 7? This win condition is a little confusing to me...